lib/appsignal/tracer.ex

```diff
@@ -258,4 +261,38 @@ defmodule Appsignal.Tracer do
       ArgumentError -> nil
     end
   end
+
+  @spec on_create_span(Span.t() | nil, Span.t() | nil) :: Span.t() | nil
+  defp on_create_span(span, parent) do
+    custom_on_create_fun =
+      Application.get_env(:appsignal, :custom_on_create_fun, &__MODULE__.custom_on_create_fun/2)
+
+    span = custom_on_create_fun.(span, parent)
```

Comment:
I think the `parent` here should be the root span, calling `Appsignal.Tracer.root_span()`, because we only support setting sample data on the root span. Otherwise this only works when you create a child span of the root span, but not any deeper nested spans.

If it is the root span, I suggest we change the variable name to be `root` or `root_span`.